models/particles/d3q27_PSM/Dynamics.c.Rt

```diff
@@ -0,0 +1,1067 @@
+                         omegaS = PV(paste("omegaS[",1:27-1,"]",sep=""))
+                         omegaF = PV("omegaF")
+
+                      C(usq, sum(u*u)*(1/c_squ))
+                      for (i in 1:27){
+                         C( feq[i],  (((u %*% t(U) * (1/c_squ))[i]) + (((u %*% t(U) * (1/c_squ))[i])*((u %*% t(U) * (1/c_squ))[i])+usq*(-1))*0.5 + 1) * d *tn[i,i])
+                      }
+	 ?>
+
+         for (FullParticleIterator p(X,Y,Z); p; ++p) {
+                
+		    real_t dx = p.diff.x;
+		    real_t dy = p.diff.y;
+		    real_t dz = p.diff.z;
+
+		    if ((DNx>0) && ((DNx - abs(dx)) < (p.rad + 2))) {
+			dx = DNx - abs(dx);
+		    }
+		    
+		    if ((DNy>0) && ((DNy - abs(dy)) < (p.rad + 2))) {
+			dy = DNy - abs(dy);
+		    }
+				
+		    if ((DNz>0) && ((DNz - abs(dz)) < (p.rad + 2))) {
+			dz = DNz - abs(dz);
+		    }
+
+		    real_t dist = sqrt(dx*dx + dy*dy + dz*dz);
+                    
+		    if ((dist - p.rad)<sqrt(3)) {
+			numP += 1;
+			uPx = ((numP - 1.0)*uPx + p.cvel.x)/numP;
+			uPy = ((numP - 1.0)*uPy + p.cvel.y)/numP;
+			uPz = ((numP - 1.0)*uPz + p.cvel.z)/numP;
+
+                    real_t localCoverage = 0.0;
+
+                    if ((dist - p.rad) < -1.0){
+                        localCoverage = 1.0;
+                    } else{
+                        localCoverage = (p.rad -  0.084/p.rad + 0.5 - dist);
+                    }
+
+                    if (localCoverage > 1.0){ localCoverage =  1.0;}
+                    if (localCoverage < 0.0){ localCoverage =  0.0;}
+
+		   
+		    sol += localCoverage;
+
+                 }
+             
+        }
+
+                    if (sol > 1.0){ sol =  1.0;}
+                    if (sol < 0.0){ sol =  0.0;}
+
+		    localOmegaF = omegaF;
+		    sol = sol*(1/localOmegaF - 0.5) / ((1 - sol) + (1/localOmegaF - 0.5));
+		    //sol = sol*((1/localOmegaF - 0.5) + 2*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5) + 3*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5) + 4*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)) / ((1 - sol) + ((1/localOmegaF - 0.5) + 2*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5) + 3*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5) + 4*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)*(1/localOmegaF - 0.5)));
+
+         for (SyncParticleIterator p(X,Y,Z); p; ++p) {
+                if ((NodeType & NODE_BOUNDARY)==0){
+		    real_t dx = p.diff.x;
+		    real_t dy = p.diff.y;
+		    real_t dz = p.diff.z;
+
+		    if ((DNx>0) && ((DNx - abs(dx)) < (p.rad + 2))) {
+			dx = DNx - abs(dx);
+		    }
+		    
+		    if ((DNy>0) && ((DNy - abs(dy)) < (p.rad + 2))) {
+			dy = DNy - abs(dy);
+		    }
+				
+		    if ((DNz>0) && ((DNz - abs(dz)) < (p.rad + 2))) {
+			dz = DNz - abs(dz);
+		    }
+
+		    real_t dist = sqrt(dx*dx + dy*dy + dz*dz);
+                    
+		    if ((dist - p.rad)<sqrt(3)) {
```

Comment:
`sqrt(3)` is disambiguous (as `3` is an integer). You should use `sqrt(3.0)`, or better yet, use:
```c++
#define SQRT_3 1.732050807568877
...
if ((dist - p.rad)<SQRT_3) {
```